src/pyhf/infer/calculators.py

```diff
@@ -155,14 +155,21 @@ def teststatistic(self, poi_test):
 
         """
         tensorlib, _ = get_backend()
-        qmu_v = qmu(poi_test, self.data, self.pdf, self.init_pars, self.par_bounds)
+
+        teststat_func = qmu_tilde if self.qtilde else qmu
+
+        qmu_v = teststat_func(
+            poi_test, self.data, self.pdf, self.init_pars, self.par_bounds
+        )
         sqrtqmu_v = tensorlib.sqrt(qmu_v)
 
         asimov_mu = 0.0
         asimov_data = generate_asimov_data(
             asimov_mu, self.data, self.pdf, self.init_pars, self.par_bounds
         )
-        qmuA_v = qmu(poi_test, asimov_data, self.pdf, self.init_pars, self.par_bounds)
+        qmuA_v = teststat_func(
```

Comment:
we probably should not call it `qmuA_v` but instead `test_statisticA_v` or similar.

Reply:
So then also renaming the class members (`sqrtqmuA_v` -> `test_stat_A_v`)?

Reply:
good point!

Reply:
@matthewfeickert which class member?

Reply:
@lukasheinrich for the calculator we have

```python
        self.sqrtqmuA_v = None
```

Reply:
I would leave this for now. The calculator is upper-limit specific right now and for asimov data (µ' = µ^) qtilde = q, We can probably address this once we get to #1009

Reply:
@lukasheinrich Yeah, I'd agree we can leave it for the time being. Hence [my comment above](https://github.com/scikit-hep/pyhf/pull/1005#discussion_r463140519).